drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -86,10 +113,21 @@ static const struct sensor_driver_api stm32_temp_driver_api = {
 static int stm32_temp_init(const struct device *dev)
 {
 	struct stm32_temp_data *data = dev->data;
+#if HAS_CALIBRATION
+	const struct stm32_temp_config *cfg = dev->config;
+#endif
 	struct adc_channel_cfg *accp = &data->adc_cfg;
 	struct adc_sequence *asp = &data->adc_seq;
 	int rc;
 
+#if HAS_CALIBRATION
+	if (!cfg->cal_vrefanalog || !cfg->cal1_addr || !cfg->cal2_addr ||
```

Comment:
This first line is superflous.
As these parameters are stated as "required" in bindings, compilation will break if one of these is not given:
```
-- Found devicetree overlay: ~/zephyrproject/zephyr/samples/sensor/stm32_temp_sensor/boards/stm32l562e_dk.overlay
devicetree error: 'ts-cal1-addr' is marked as required in 'properties:' in ~/zephyrproject/zephyr/dts/bindings/sensor/st,stm32-temp-cal.yaml, but does not appear in <Node /stm-temp in '~/zephyrproject/zephyr/misc/empty_file.c'>

```

Reply:
I tried to answer this one in your previous remark on this line:

> you still can assign 0 to all of these values in the dts, which would cause division by 0 or null pointer reference. The checks are not covered by dt tools, or at least I couldn't find a range option in the dts bindings, so I think it's safer to keep them if not possible in the dts binding?

So this should cover a dts config as
```
/ {
    stm-temp {
        label = "DIE_TEMP";
        compatible = "st,stm32-temp-cal";
        io-channels = <&adc1 17>;
        status = "okay";
        ts-cal1-addr = <0>;
        ts-cal2-addr = <0>;
        ts-cal1-temp = <30>;
        ts-cal2-temp = <110>;
        ts-cal-vrefanalog = <0>;
        ts-cal-offset = <30>;
    };
};
```

I will remove it since it's less likely to make mistakes on these now, since it is moved to the soc dts instead of the user's dts.